game_logic.cpp

```diff
@@ -0,0 +1,118 @@
+#include "game_logic.h"
+
+//gets input from console if player is human
+//generates random input if player is bot
+std::string getPlayerInput(Board &board, Player player, int askCoordinate = 0){
+    std::string ret = "";
+    if(player.isAI == false){
+        std::cin >> ret;
+        return ret;
+    }
+    else{
+        std::vector<Position> validTiles = getValidTiles(board); //TODO: make bot not be able to choose the positions where their own mines are placed
+        int randomPos = rand() % validTiles.size();
+        if(askCoordinate == 0){
+            ret = std::to_string(validTiles[randomPos].xpos);
+        }
+        else{
+            ret = std::to_string(validTiles[randomPos].ypos);
+        }
+        std::cout << std::endl;
+    }
+    return ret;
+}
+
+void chooseMinePositions(Board &board, Player &player){
+    for(int m = 0; m < player.mineCount; m++){
+        printBoard(board, player.id);
+        bool validPlacement = false;
+        Mine mine;
+        while(validPlacement == false){
+            std::cout << "Player " << player.id + 1 << "!, choose your mine's positions" << std::endl;
+            std::cout << "Choose the x position of mine " << m+1 << ": ";
+            mine.xpos = std::stoi(getPlayerInput(board, player, 0));;
+            std::cout << "Choose the y position of mine " << m+1 << ": ";
+            mine.ypos = std::stoi(getPlayerInput(board, player, 1));;
+            mine.owner = player.id;
+            validPlacement = isPositionValid(board, mine);
+            if(validPlacement==false){
+                std::cout << "Invalid position! Choose again" << std::endl;
+            }
+            for(std::vector<Mine>::iterator it = board.placedMines.begin(); it!= board.placedMines.end(); it++){
+                if (*it == mine){ //for them to be equal, they need to share position and owner
+                    std::cout << "You already placed a mine there! Choose again" << std::endl;
+                    validPlacement = false;
+                }
+            }
+            
+        }
+        //std::cout << "Player " << mine.owner+1 <<  " placed mine at " << mine.xpos << ", " << mine.ypos << std::endl; //for testing purposes, TODO: remove before main release
+        board.placedMines.push_back(mine);
+        std::cout << std::endl;
+        system("cls");
+
+    }
+}
+
+void guess(Board &board, Player player){
+    printBoard(board, player.id);
+    Position guess;
+    int isGuessValid = false; //flag to check if the inputed position is valid. If not, ask the player again
+    while(isGuessValid == false){
+        std::cout << "Player " << player.id+1 <<  ", take a guess... " << std::endl;
+        std::cout << "X position: ";
+        guess.xpos = std::stoi(getPlayerInput(board, player, 0));;
+        std::cout << std::endl;
+        std::cout << "Y position: ";

```

Comment:
#could do this instead of using two different calls: `std::cout << "\nY position:";`